plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue

```diff
@@ -0,0 +1,260 @@
+<!--
+  Matomo - free/libre analytics platform
+  @link https://matomo.org
+  @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+-->
+
+<template>
+  <div class="pluginSettings" ref="root">
+    <div
+      class="card"
+      v-for="settings in settingsPerPlugin"
+      :id="`${settings.pluginName}PluginSettings`"
+      :key="`${settings.pluginName}PluginSettings`"
+    >
+      <div class="card-content">
+        <h2
+          class="card-title"
+          :id="settings.pluginName"
+        >{{ settings.title }}</h2>
+        <div
+          v-for="setting in settings.settings"
+          :key="`${setting.pluginName}.${setting.name}`"
+        >
+          <div>
+            <FormField
+              v-model="settingValues[`${settings.pluginName}.${setting.name}`]"
+              :form-field="{
+                ...setting,
+                condition: makeSettingConditionFunction(setting, settings.pluginName),
+              }"
+            />
+          </div>
+        </div>
+        <input
+          type="button"
+          @click="save(settings.pluginName)"
+          :disabled="isLoading"
+          class="pluginsSettingsSubmit btn"
+          :value="translate('General_Save')"
+        />
+        <ActivityIndicator
+          :loading="isLoading || isSaving[settings.pluginName]"
+        />
+      </div>
+    </div>
+    <div class="confirm-password-modal modal">
+      <div class="modal-content">
+        <h2>{{ translate('UsersManager_ConfirmWithPassword') }}</h2>
+        <div>
+          <Field
+            v-model="passwordConfirmation"
+            :uicontrol="'password'"
+            :name="'currentUserPassword'"
+            :autocomplete="false"
+            :full-width="true"
+            :title="translate('UsersManager_YourCurrentPassword')"
+          >
+          </Field>
+        </div>
+      </div>
+      <div class="modal-footer">
+        <a
+          href=""
+          class="modal-action modal-close btn"
+          :disabled="!passwordConfirmation ? 'disabled' : undefined"
+          @click="$event.preventDefault(); save(this.settingsToSave)"
+        >{{ translate('General_Yes') }}</a>
+        <a
+          href=""
+          class="modal-action modal-close modal-no"
```

Comment:
Wondering if adding that might make sense, as otherwise the page reloads when clicking on "no"
```suggestion
          class="modal-action modal-close modal-no"
          @click="$event.preventDefault();"
```
Or maybe that needs to be fixed in the modal, as it wasn't the case before 🤔